pilot/pkg/serviceregistry/cloudfoundry/servicediscovery.go

```diff
@@ -67,9 +67,8 @@ func (sd *ServiceDiscovery) Services() ([]*model.Service, error) {
 	}
 
 	internalRouteServicePort := &model.Port{
-		Port:     sd.ServicePort,
-		Protocol: model.ProtocolTCP,
-		Name:     "tcp",
+		Port: sd.ServicePort,
+		Name: "tcp",
```

Comment:
this essentially nullifies the change you put in above. You need to set the protocol as HTTP if you want to use downstream protocol.. and preferably use name http-xx

Reply:
This was just for internal routes (which are all TCP). We left the protocol on the routes that are used for external routes. Appreciate you pointing this out.